be/src/exec/vectorized/hdfs_scan_node.cpp

```diff
@@ -392,6 +389,12 @@ void HdfsScanNode::_scanner_thread(HdfsScanner* scanner) {
     ChunkPtr chunk = nullptr;
 
     while (status.ok()) {
+        // if global status was not ok, we need fast failure and no need to read file
+        if (!_get_status().ok()) {
+            _release_scanner(scanner);
```

Comment:
Could you refactor it to `break`?  And following code will handle `_get_status().ok()`.  I want to the whole workflow simple as possible.